rad/prelude/sequence.rad

```diff
@@ -0,0 +1,12 @@
+{:module  'prelude/sequence
+ :doc     "Functions which work on both lists and vectors."
+ :exports '[concat]}
+
+(import prelude/basic :unqualified)
```

Comment:
Some of the stuff in `basic.rad` should probably be moved here (`length`, `some`, `elem?`, `empty-seq?`).

Reply:
Some experimentation shows that we can re-shuffle `prelude/basic` and `prelude/list` quite a bit, moving some into `prelude/sequence`, but I'll leave this to another PR (after #277 is merged).

Reply:
Okay